File: number-of-islands/njngwn.java
// Time Complexity: O(m*n), m: number of row, n: number of column
// Space Complexity: O(m*n), m: number of row, n: number of column, because of call stack
class Solution {
    void findLand(char[][] grid, int row, int col) {
        if (row < 0 || row > grid.length-1 || col < 0 || col > grid[0].length-1) return;
        if (grid[row][col] == '0') return;

        grid[row][col] = '0';   // make element '0'(water)

        findLand(grid, row-1, col);
        findLand(grid, row+1, col);
        findLand(grid, row, col-1);
        findLand(grid, row, col+1);
    }

    public int numIslands(char[][] grid) {
        int num = 0;

        for (int i = 0; i < grid.length; ++i) {
            for (int j = 0; j < grid[0].length; ++j) {
                if (grid[i][j] == '1') {
                    findLand(grid, i, j);
                    num++;
                }
            }
        }

        return num;
    }
}

File: reverse-bits/njngwn.py
class Solution:
    # Time Complexity: O(1)
    # Space Complexity: O(1)
    def reverseBits(self, n: int) -> int:
        stack = []
        for i in range(32):
            stack.append(n % 2)
            n //= 2

        res, multiples = 0, 1
        while stack:
            res += (stack.pop() * multiples)
            multiples *= 2

        return res
